Stratis.Bitcoin/BlockPulling/DownloadAssignmentStrategy.cs

```diff
@@ -0,0 +1,93 @@
+﻿using NBitcoin.Protocol;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using System.Collections.Concurrent;
+using NBitcoin.Protocol.Behaviors;
+using System.Threading;
+using Stratis.Bitcoin.Connection;
+
+namespace Stratis.Bitcoin.BlockPulling
+{
+    /// <summary>
+    /// Implements a strategy for a block puller that needs to download 
+    /// a set of blocks and from its connected peers.
+    /// </summary>
+    public static class DownloadAssignmentStrategy
```

Comment:
I am not sure about naming the pattern in to the class, maybe a more suitable option is to include Puller like PullerDownloadAssignments or PullerDistributionAssignments.